.github/workflows/reusable-retry-buildkite-step-on-events.yml

```diff
@@ -0,0 +1,123 @@
+on:
+  workflow_call:
+    inputs:
+      org-slug:
+        description: 'Buildkite organization slug'
+        required: true
+        type: string
+      pipeline-slug:
+        description: 'Slug of the Buildkite pipeline to be run'
+        required: true
+        type: string
+      retry-step-key:
+        description: 'Key of the Buildkite job to be retried'
+        required: true
+        type: string
+      build-commit-sha:
+        description: 'Commit to check for running Buildkite Builds on. Usually github.event.pull_request.head.sha .'
+        required: true
+        type: string
+      cancel-running-github-jobs:
+        description: 'Cancel currently in progress Github jobs when new ones are added.'
+        default: true
+        type: boolean
+        required: false
+    secrets:
+      buildkite-api-token:
+        required: true
+
+concurrency:
+  group: danger-buildkite-retry-${{ github.ref }}
+  cancel-in-progress: ${{ inputs.cancel-running-github-jobs }}
+
+jobs:
+  retry-buildkite-job:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "🔄 Retry job on the latest Buildkite Build"
+        run: |
+          ORG_SLUG="${{ inputs.org-slug }}"
+          PIPELINE_SLUG="${{ inputs.pipeline-slug }}"
+          RETRY_STEP_KEY="${{ inputs.retry-step-key }}"
+          BUILDKITE_API_ACCESS_TOKEN="${{ secrets.buildkite-api-token }}"
+          COMMIT_SHA="${{ inputs.build-commit-sha }}"
+
+          # Performs a Buildkite request using a http method ($1) and an api path ($2)
+          perform_buildkite_request() {
+            local METHOD=$1
+            local BUILDKITE_API_PATH=$2
+
+            local BUILDKITE_API_URL="https://api.buildkite.com/v2/organizations/$ORG_SLUG/pipelines/$PIPELINE_SLUG/$BUILDKITE_API_PATH"
+
+            local RAW_RESPONSE=$(
+              curl \
+                --fail-with-body \
+                --silent \
+                --show-error \
+                -X "$METHOD" \
+                -H "Authorization: Bearer $BUILDKITE_API_ACCESS_TOKEN" \
+                "$BUILDKITE_API_URL"
+            )
+
+            echo "$RAW_RESPONSE" | tr -d '\n' | jq -R -r
```

Comment:
What's the goal of using `-R -r` here? 🤔

Reply:
Also wouldn't the `tr` command risk removing newlines also within strings which are part of the response (thus modifying the response content itself, not just its formatting)? (e.g. `{ "BUILDKITE_MESSAGE": "Hello\nWorld" }`)

Reply:
Thanks for pointing those out -- tbh, I can't remember why I originally had those 🙃. Perhaps the `tr` was helpful when developing?
They didn't seem to break the result but you're right, they're not necessary. I've fixed a few other shellcheck issues together with these on ec6dca9257a68818aeaf82741bff0d4f09f577a0.